### PR TITLE
[PICA-102] add claimed_amount to Claimed event

### PR DESCRIPTION
**Pull request by kkast**
https://app.clickup.com/t/86775ywct

- add claimed_amount to Claimed event
- removed logic for slashing rewards for early unstakes
- fixed tests

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [PICA-102 Add claimed_amount to claimed event](https://app.clickup.com/t/20465559/PICA-102)

**Comment by kkast** [comment]
@benluelo please take a look at collect_rewards function logic

**Review by vimukthi-git — APPROVED** [review]

File: code/parachain/frame/staking-rewards/src/lib.rs [hunk 1]
```diff
@@ -167,6 +167,8 @@ pub mod pallet {
 			fnft_collection_id: T::AssetId,
 			/// FNFT Instance Id
 			fnft_instance_id: T::FinancialNftInstanceId,
+			// Claimed amount map<reward id, amount>
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/lib.rs:170]: This comment doesn't add anything, can be removed

File: code/parachain/frame/staking-rewards/src/lib.rs [hunk 2]
```diff
@@ -167,6 +167,8 @@ pub mod pallet {
...
+			claimed_amount: BTreeMap<T::AssetId, T::Balance>,
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/lib.rs:171]: `claimed_amounts` makes more sense here, as there is more than one amount

File: code/parachain/frame/staking-rewards/src/test/mod.rs [hunk 3]
```diff
@@ -1378,6 +1378,7 @@ fn claim_with_insufficient_pot_funds() {
 				owner: BOB,
 				fnft_collection_id: STAKING_FNFT_COLLECTION_ID,
 				fnft_instance_id: bob_stake_id,
+				claimed_amount: BTreeMap::from([(USDT::ID, 900000)]),
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/test/mod.rs:1381]: would be helpful to explain where these claimed amounts come from, or use a function to get the value (i.e. claim_of_stake or something similar)

File: code/parachain/frame/staking-rewards/src/test/test_update_reward_pools.rs [hunk 4]
```diff
@@ -217,6 +218,7 @@ fn update_accumulates_properly() {
 				owner: CHARLIE,
 				fnft_collection_id: STAKING_FNFT_COLLECTION_ID,
 				fnft_instance_id: stake_id,
+				claimed_amount: BTreeMap::from([(USDT::ID, USDT::units(1) / 1000 * 6)]),
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/test/test_update_reward_pools.rs:221]: prefer to use `[..].into_iter().collect()` over `BTreeMap::from([..])`

Comment (benluelo) [reply code/parachain/frame/staking-rewards/src/test/test_update_reward_pools.rs:221]: There is also a `btree_map` helper function that can be used as well, but .iter.collect tends to format more nicely

**Review by benluelo — CHANGES_REQUESTED** [review]

File: code/parachain/frame/staking-rewards/src/test/mod.rs (hunk 3 above)
Comment (benluelo) [reply code/parachain/frame/staking-rewards/src/test/mod.rs:1381]: also `900000` should be `900_000`

**Review by benluelo — APPROVED** [review]

**Pull request merged**